x/evm/keeper/state_transition.go

```diff
@@ -354,7 +354,7 @@ func (k *Keeper) ApplyMessageWithConfig(ctx sdk.Context,
 			return nil, errorsmod.Wrap(err, "failed to apply state override")
 		}
 	}
-
+	ctx = ctx.WithValue(statedb.StateDBContextKey, stateDB)
```

Comment:
It's already done in state db constructor.

Reply:
seems still have issue, we can't modify ctx outside

Reply:
hooks?

Reply:
hooks?

Reply:
what issues?

Reply:
what issues?

Reply:
we can't get stateDb [ctx](https://github.com/crypto-org-chain/cronos/blob/9cb8be65e2adf5fc392d584ebff1774d4387c841/x/cronos/keeper/evm.go#L42) without using relayer precompile?

Reply:
no, https://github.com/crypto-org-chain/cronos/pull/1233 don't need to change relayer precompile.

Reply:
no, https://github.com/crypto-org-chain/cronos/pull/1233 don't need to change relayer precompile.

Reply:
no log [without](https://github.com/crypto-org-chain/cronos/pull/1233/commits/25c3feedfb8e85b2509f82a39cc0743e7c778a2d) using relayer precompile config

Reply:
no log [without](https://github.com/crypto-org-chain/cronos/pull/1233/commits/25c3feedfb8e85b2509f82a39cc0743e7c778a2d) using relayer precompile config